firedrake/slate/slac/tsfc_driver.py

```diff
@@ -47,10 +60,12 @@ def compile_terminal_form(tensor, prefix=None, tsfc_parameters=None):
     cxt_kernels = []
     for orig_it_type, integrals in transformed_integrals.items():
         subkernel_prefix = prefix + "%s_to_" % orig_it_type
-        kernels = tsfc_compile(Form(integrals),
+        form = Form(integrals)
+        kernels = tsfc_compile(form,
                                subkernel_prefix,
                                parameters=tsfc_parameters)
         cxt_k = ContextKernel(tensor=tensor,
```

Comment:
I note in passing that this is weird.  I would have expected that for each tensor there is one and only one `ContextKernel`, but it appears not.  There are `len(transformed_integrals)` `ContextKernel`s, all sharing the same `tensor`